### PR TITLE
added missing math functions from intel's libm

### PR DESCRIPTION
**Pull request by Goirad**

### CODE REVIEW

File: rs-libc/src/asm/x86_64/sqrt.S [hunk 1]
```diff
@@ -0,0 +1,80 @@
+/*
```
Comment (jethrogb) [inline rs-libc/src/asm/x86_64/sqrt.S]: Is this file necessary? That's surprising to me. What happens if you call `f64::sqrt` in Rust?

Comment (Goirad) [reply rs-libc/src/asm/x86_64/sqrt.S]: I'll have to try, but it was up there with the other missing math functions.

File: rs-libc/LICENSE [hunk 2]
```diff
@@ -29,6 +29,21 @@ Adapted from musl-libc (license below):
     src/c/strtod.c
     src/c/strtol.c
 
+Adapted from Intel's linux-sgx libm reference (license in files):
```
Comment (jethrogb) [inline rs-libc/LICENSE]: Can you actually extract the license into this file (like musl below)?

Comment (jethrogb) [inline rs-libc/LICENSE]: Remove the word "reference"

**Comment by jethrogb** [comment]
r=me after squashing all commits into one

**Comment by Goirad** [comment]
Commits have been squashed
